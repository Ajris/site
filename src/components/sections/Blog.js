import React from 'react';
import CardDeck from "react-bootstrap/CardDeck";
import ItemCard from "./ItemCard";

export default class Blog extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            cards: [],
        };
    }

    componentDidMount() {
        fetch("http://localhost:8080/blog")
            .then(res => res.json())
            .then((result) => {
                let i = 0;
                let cards = result.map((item) =>
                    <ItemCard key={i++} title={item.name} text={item.email} footer={"footer"}/>
                );
                this.setState({cards:cards});
            });

    }

    render() {
        return (
            <div>
                <div className={'mx-4 my-3'}>
                    <p className={'topic text-center'}>Blog</p>
                </div>
                <CardDeck>
                    {this.state.cards}
                </CardDeck>
            </div>
        )
    }
}