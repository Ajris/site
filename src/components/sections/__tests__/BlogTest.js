import {create} from "react-test-renderer";
import React from "react";
import Blog from "../Blog";

describe("Blog component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Blog/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});