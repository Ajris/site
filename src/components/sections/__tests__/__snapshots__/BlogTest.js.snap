// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog component it matches the snapshot 1`] = `
<div>
  <div
    className="mx-4 my-3"
  >
    <p
      className="topic text-center"
    >
      Blog
    </p>
  </div>
  <div
    className="card-deck"
  />
</div>
`;
